app/assets/javascripts/discourse/tests/acceptance/user-menu-test.js

```diff
@@ -68,22 +92,64 @@ acceptance("User menu - Dismiss button", function (needs) {
     await click(".user-menu .notifications-dismiss");
     assert.strictEqual(
       query(".dismiss-notification-confirmation").textContent.trim(),
-      I18n.t("notifications.dismiss_confirmation.body", { count: 10 }),
+      I18n.t("notifications.dismiss_confirmation.body.default", { count: 10 }),
       "confirmation modal is shown when there are unread high pri notifications"
     );
     assert.notOk(markRead, "mark-read request isn't sent");
 
     await click(".modal-footer .btn-default"); // click cancel on the dismiss modal
 
-    updateCurrentUser({ unread_high_priority_notifications: 0 });
+    await publishToMessageBus(`/notification/${loggedInUser().id}`, {
+      unread_high_priority_notifications: 0,
+    });
     await click(".user-menu .notifications-dismiss");
     assert.ok(
       markRead,
       "mark-read request is sent without a confirmation modal when there are no unread high pri notifications"
     );
   });
 
-  test("doesn't show confirmation modal for the likes notifications panel/list", async function (assert) {
+  test("shows confirmation modal for the bookmarks list", async function (assert) {
+    await visit("/");
+    await click(".d-header-icons .current-user");
+
+    assert.strictEqual(
+      query("#user-menu-button-bookmarks .badge-notification").textContent,
+      "103",
+      "bookmarks tab has bubble with count"
+    );
+
+    await click("#user-menu-button-bookmarks");
+    await click(".user-menu .notifications-dismiss");
+
+    assert.strictEqual(
+      query(".dismiss-notification-confirmation").textContent.trim(),
+      I18n.t("notifications.dismiss_confirmation.body.bookmarks", {
+        count: 103,
+      }),
+      "confirmation modal is shown when there are unread bookmark reminder notifications"
+    );
+    assert.ok(!markRead, "mark-read request isn't sent");
+
+    await click(".modal-footer .btn-primary"); // confirm dismiss on the dismiss modal
+    await publishToMessageBus(`/notification/${loggedInUser().id}`, {
+      grouped_unread_high_priority_notifications: {},
+    });
```

Comment:
Why do we need a messageBus message to clear the count? I would have thought that we would clear the count after the `mark-read` request.

Reply:
Because there are several things that can change a notification's read status (e.g. when you click on a topic that has unread notifications they're all automatically marked read) so we just rely on message bus entirely to update the current user state.

Reply:
> so we just rely on message bus entirely to update the current user state.

That might not be ideal because of the potential delays with MessageBus. We should just clear the count when making the requests.